### PR TITLE
Fix a bug in completion if callback value is not converted to string

### LINKED ISSUE #2223: Spyder2.3.3 Code completion breaks
Hi everyone,

first of all, a big thank you for developing this great spyder IDE. It is by far my weapon of choice and I enjoyed diving into python a lot more thanks to it. 

Now to my problem, which I am assuming is on my machine, but I cannot figure it out so I hope you guys can help me:

I just installed spyder 2.3.3 on a fresh ubuntu trusty installation and also installed jedi and rope with pip.

However, when I try to use auto completion (e.g. for the module time) I get the following error message from the internal console:

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/spyderlib/utils/introspection/plugin_manager.py", line 58, in _handle_incoming
    self._threads[name].wait()
KeyError: PyQt4.QtCore.QString(u'rope')

I already tried to debug this error myself by adding basic "prints" in the file, but in my oppionion the key should be present. What I did find however was that I had to use the python3 print with the (). Is that something you forced, or does my system try to use python3 for some reason?

I hope this is the correct place to state such an individual problem and maybe someone can help me.

Edit: I just tried updating my spyder in my VM running Ubuntu precise and I get the same error. I installed spyder2.3.3 from source there since the pip installer would not work (threw the error:
error: option --single-version-externally-managed not recognized).

Am I missing something that I must do before code completion works?

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #2223.

### CODE REVIEW

File: spyderlib/utils/introspection/plugin_manager.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def _handle_timeout(self):
         self.waiting = False
 
     def _handle_incoming(self, name):
+        # coerce to a str in case it is a QString
+        name = str(name)
```
Comment (ccordoba12) [inline spyderlib/utils/introspection/plugin_manager.py]: Is this Python 2 and 3 compatible?

Comment (blink1073) [inline spyderlib/utils/introspection/plugin_manager.py]: It is now.  :smile:

Comment (ccordoba12) [inline spyderlib/utils/introspection/plugin_manager.py]: But you're getting bytes in PY2 and unicode in PY3, is that ok in this case?

Comment (blink1073) [inline spyderlib/utils/introspection/plugin_manager.py]: Its okay in that its consistent, and it works

Comment (ccordoba12) [inline spyderlib/utils/introspection/plugin_manager.py]: Ok, trusting you on this one :-)

**Pull request merged**